app/dependencies.py

```diff
@@ -139,6 +143,40 @@ def get_redis_client(
     return redis_client
 
 
+def get_notification_manager(
+    settings: Settings = Depends(get_settings),
+) -> NotificationManager:
+    """
+    Dependency that returns the notification manager
+    """
+    global notification_manager
+
+    if notification_manager is None:
+        notification_manager = NotificationManager(settings=settings)
+
+    return notification_manager
+
+
+def get_notification_tool(
+    background_tasks: BackgroundTasks,
+    settings: Settings = Depends(get_settings),
+    db: AsyncSession = Depends(get_db),
+) -> NotificationTool:
+    """
+    Dependency that returns the notification manager
+    """
+    global notification_manager
```

Comment:
Can't you just call get_notification_manager ?

Reply:
This is a really pertinent idea. Done with [eaea40d](https://github.com/aeecleclair/Hyperion/pull/178/commits/eaea40d992caa8d23636799b081c73e066b00937)